### PR TITLE
Add Memento

### PR DESCRIPTION
**Pull request by silverweed**
Adding the Memento pattern. Code is formatted and runs correctly.

### CODE REVIEW

File: behavioral/memento.cr [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  def initialize(fighter1, fighter2, rounds, time)
+    @state = State.new({fighter1, fighter2}, rounds, time)
+  end
+
+  def saveMemento
+    @state
+  end
+
+  def restoreMemento(memento : Match::State)
+    @state = memento
```
Comment (veelenga) [inline behavioral/memento.cr]: shouldn't this be `save_memento` and `restore_memento` ?

Comment (silverweed) [reply behavioral/memento.cr]: Yeah, definitely. It was a camelCase-esque lapsus from my part. Fixing it as soon as I have a PC available :+1:

**Comment by silverweed** [comment]
Fixed wrong case for variables and methods

**Comment by veelenga** [comment]
Thanks 👍

**Pull request merged**